### PR TITLE
First pass at improving reverseWords question.

### PR DESCRIPTION
**Pull request by seanlip**
This is a first pass at adding more feedback to the reverseWords question. There's yet more feedback that can be added, but I'm blocked by a lack of functionality, so I figured I'd submit this first, add the missing functionality, and then do another pass at adding more feedback.

I also found and fixed a couple of existing bugs (@theresa-lee PTAL):
- hintIndex was not being carried over when the code received "don't use print statements" feedback, so only the first message of a feedback set was ever being shown. This resulted in a frustrating UX since the later feedback was never reached.
- The loading dots were overlapping the reinforcements, which looked ugly. I gave them a top margin.

The additional functionality I plan to add in future PRs is the following:
- After giving a sequence of hints, show "you output X but we expected Y", so at least it's clear exactly which input is failing.
- Allow custom feedback to be provided, based on which _test suites_ passed/failed.
- Allow buggy output tests to be run on only a subset of the test suites. This is because the code could fail in one way for some of the "easier" test suites, which allows appropriate feedback to be given -- but then fail in multiple ways for the "corner case" tests depending on how buggy the implementation is. This makes it hard to write custom buggy functions to catch such errors, since currently all buggy functions run on all test suites.

### CODE REVIEW

File: assets/questions/reverseWords.js [hunk 1]
```diff
@@ -28,124 +28,388 @@ globalData.questions['reverseWords'] = {  // eslint-disable-line dot-notation
...
+          'Adding spaces at ends of words to keep things simple is a nice ',
+          'idea, but be careful about how you handle the extra space at the ',
+          'end.'
+        ].join('')
+      ]
+    }, {
+      buggyFunctionName: 'AuxiliaryCode.treatAsciiCharsContiguously',
+      messages: [
+        [
+          'Check the ASCII tables -- are you sure that "A"-"z" be treated as ',
```
Comment (theresa-lee) [inline assets/questions/reverseWords.js]: Should 'be' -> is/are?

File: assets/questions/reverseWords.js [hunk 2]
```diff
@@ -28,124 +28,388 @@ globalData.questions['reverseWords'] = {  // eslint-disable-line dot-notation
...
+          'a contiguous block of characters?'
+        ].join('')
+      ]
+    }, {
+      buggyFunctionName: 'AuxiliaryCode.useWeakInequalitiesForCharValidity',
+      messages: [
+        [
+          'Are you correctly handling the check for whether a character is a ',
```
Comment (theresa-lee) [inline assets/questions/reverseWords.js]: Does there need to be an "or not" in here? I can't tell, it just feels like it's missing for me

File: assets/questions/reverseWords.js (hunk 1 above)
Comment (seanlip) [reply assets/questions/reverseWords.js]: Thanks, good catch! Fixed.

File: assets/questions/reverseWords.js (hunk 2 above)
Comment (seanlip) [reply assets/questions/reverseWords.js]: I checked -- consensus [here](https://english.stackexchange.com/questions/3382/whether-or-not-vs-whether) seems to be to just use "whether", so I'm going to go for that.
